runtime/stdlib/cadence_v0.42_to_v1_contract_upgrade_validator.go

```diff
@@ -538,12 +540,27 @@ func (validator *CadenceV042ToV1ContractUpdateValidator) checkConformanceV1(
 	newDecl *ast.CompositeDeclaration,
 ) {
 
-	// Here it is assumed enums will always have one and only one conformance.
+	oldConformances := oldDecl.Conformances
+
+	// NOTE 1: Here it is assumed enums will always have one and only one conformance.
 	// This is enforced by the checker.
-	// Therefore, below check for multiple conformances is only applicable
-	// for non-enum type composite declarations. i.e: structs, resources, etc.
+	//
+	// NOTE 2: If one declaration is an enum, then other is also an enum at this stage.
+	// This is enforced by the validator (in `checkDeclarationUpdatability`), before calling this function.
+	if newDecl.Kind() == common.CompositeKindEnum {
+		err := oldConformances[0].CheckEqual(newDecl.Conformances[0], validator)
```

Comment:
Is the update checker always only run if checking of the new (and old?) contract succeeded? If not, maybe still check the lengths of the conformances before indexing.

Reply:
existing code should have been type checked already when deploying. The new code gets type checked before running the validator. So the size should already be validated for both programs.
Can still add a check though, just to be safe.